workloads/librispeech/librispeech_jax/workload.py

```diff
@@ -0,0 +1,286 @@
+        "A": 3,
+        "B": 4,
+        "C": 5,
+        "D": 6,
+        "E": 7,
+        "F": 8,
+        "G": 9,
+        "H": 10,
+        "I": 11,
+        "J": 12,
+        "K": 13,
+        "L": 14,
+        "M": 15,
+        "N": 16,
+        "O": 17,
+        "P": 18,
+        "Q": 19,
+        "R": 20,
+        "S": 21,
+        "T": 22,
+        "U": 23,
+        "V": 24,
+        "W": 25,
+        "X": 26,
+        "Y": 27,
+        "Z": 28,
+    }
+    self._rev_label_dict = {v: k for k, v in self._label_dict.items()}
+    self._decoder = ctcdecode.CTCBeamDecoder(
+        labels=[str(c) for c in self._rev_label_dict], beam_width=1)
+    self._loss = ctc_loss.ctc_loss
+  
+  def has_reached_goal(self, eval_result: float) -> bool:
+    return eval_result < self.target_value
+  
+  def build_input_queue(self, data_rng, split: str, data_dir: str,
+                        batch_size: int):
+    torch.manual_seed(data_rng[0])
+    train_set = input_pipeline.LibriSpeechDataset(
+        os.path.join(data_dir, "features_train-clean-100.csv"))
+    valid_set = input_pipeline.LibriSpeechDataset(
+        os.path.join(data_dir, "features_test-clean.csv"))
+
+    train_collate_fn = train_set.pad_collate
+
+    self._train_loader = torch.utils.data.DataLoader(
+        train_set,
+        batch_size=batch_size,
+        shuffle=True,
+        num_workers=2,
+        pin_memory=True,
+        collate_fn=train_collate_fn)
+
+    self._valid_loader = torch.utils.data.DataLoader(
+        valid_set,
+        batch_size=batch_size,
+        num_workers=2,
+        pin_memory=True,
+        collate_fn=train_collate_fn)
+
+    return iter(itertools.cycle(self._train_loader))
+
+  def sync_batch_stats(self, model_state):
+    """Sync the batch statistics across replicas."""
+    # An axis_name is passed to pmap which can then be used by pmean.
+    # In this case each device has its own version of the batch statistics and
+    # we average them.
+    avg_fn = jax.pmap(lambda x: lax.pmean(x, 'x'), 'x')
+    new_model_state = model_state.copy({
+      'batch_stats': avg_fn(model_state['batch_stats'])})
+    return new_model_state    
+
+  @property
+  def param_shapes(self):
+    if self._param_shapes is None:
+      raise ValueError(
+          'This should not happen, workload.init_model_fn() should be called '
+          'before workload.param_shapes!')
+    return self._param_shapes
+  
+  @property
+  def target_value(self):
+    return 0.1
+
+  @property
+  def loss_type(self):
+    return spec.LossType.CTC_LOSS
+
+  @property
+  def num_train_examples(self):
+    return 28539
+
+  @property
+  def num_eval_examples(self):
+    return 2620
+
+  @property
+  def train_mean(self):
+    return 0.0
+
+  @property
+  def train_stddev(self):
+    return 1.0
+  
+  def model_params_types(self):
+    pass
+
+  @property
+  def max_allowed_runtime_sec(self):
+    return 80000
+
+  @property
+  def eval_period_time_sec(self):
+    return 800
```

Comment:
all of these properties should be constant between pytorch and jax, would it be possible for you to make a parent `LibriSpeechWorkload` abstract class in `workloads/librispeech` that defines these, and then have both the Jax and Pytorch subclasses inherit from it?